src/Parser.php

```diff
@@ -71,9 +69,10 @@ public function isScheme(string $scheme): bool
     public function isHost(string $host): bool
     {
         return '' === $host
+            || $this->isRegisteredName($host)
             || filter_var($host, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4)
```

Comment:
FWIW ... a IPv4 host is a "special registered name" so for a URI parser point of view there's no necessity to distinguish between a registered name and an IPv4 host. this check can be safely remove